test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+				// Thus, this is only an artificial assignment of EgressIP to node on these cloud platforms and the EgressIP feature
+				// will pick the actual node.
+				g.By("Getting a map of source nodes and potential Egress IPs for these nodes")
+				egressIPsPerNode := 1
+				nodeEgressIPMap, err := findNodeEgressIPs(oc, egressIPNodeStr, cloudType, egressIPsPerNode)
+				framework.Logf("%v", nodeEgressIPMap)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Choosing the EgressIPs to be assigned, one per node")
+				var egressIPs []string
+				egressIPSet := make(map[string]struct{})
+				for _, eip := range nodeEgressIPMap {
+					_, ok := egressIPSet[eip[0]]
+					if !ok {
+						egressIPs = append(egressIPs, eip[0])
+						egressIPSet[eip[0]] = struct{}{}
+					}
+				}
+
+				g.By("Marshalling the desired EgressIPs into a string")
+				egressIPsString, err := json.Marshal(egressIPs)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Creating the EgressIP object and writing it to disk")
+				var egressIPConfig = fmt.Sprintf(
+					egressIPYamlTemplateNamespaceSelector, // template yaml
+					egressIPNamespace,                     // name of EgressIP
+					egressIPsString,                       // compact yaml of egressIPs
+					fmt.Sprintf("kubernetes.io/metadata.name: %s", egressIPNamespace), // namespace selector
+				)
+				err = ioutil.WriteFile(tmpDirEgressIP+"/"+egressIPYaml, []byte(egressIPConfig), 0644)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Applying the EgressIP object")
+				_, err = oc.AsAdmin().Run("create").Args("-f", tmpDirEgressIP+"/"+egressIPYaml).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// This approach here is different from the other tests because:
+				// a) No additional SNAT or similar can be injected by the cloud as we go directly from node and we know that we have
+				// an endpoint on the cloud, always, thus we can directly query agnhost's /clientip.
+				// b) The requests in tcpdump did not expose the request string for some reason (probably needed better filters)
+				// c) It's simpler to just query for the /clientip instead of relying on the packet capture for these tests.
+				for _, routeName := range routeNames {
+					g.By(fmt.Sprintf("Launching a new prober pod and probing for EgressIPs at %s", routeName))
+					iterations := 10
+					clientIPSet, err := probeForClientIPs(oc, externalNamespace, probePodName, routeName, targetIP, targetPort, iterations)
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					g.By("Making sure that EgressIPs were not part of the response")
+					framework.Logf("egressIPSet is: %v", egressIPSet)
+					framework.Logf("clientIPSet is: %v", clientIPSet)
+					o.Expect(len(clientIPSet)).Should(o.BeNumerically(">", 0))
+					// Note: it's a bug if we see an egressIP here:
+					// https://bugzilla.redhat.com/show_bug.cgi?id=2070929
```

Comment:
this bug is still marked as assigned (not fixed), so does this fail every time? I saw in your comments that most of these tests are PASS.

Reply:
So to begin with, I will disable those internal tests because they fail until 2070929 is fixed. As soon as Surya fixes that bug, I will talk to her and Tim what to do with this here (I updated the comment for clarification):
~~~
269                                         // Note: my interpretation is that it's a bug if we see an egressIP here:
270                                         // We should never see egressIPs when querying internal targets:
271                                         // https://bugzilla.redhat.com/show_bug.cgi?id=2070929
272                                         // However, this was still a subject of discussion. When we enable these tests after
273                                         // we fix 2070929, decide if we want to see EgressIPs here or not and possibly remove
274                                         // this verification.
275                                         g.By("Making sure that EgressIPs were not part of the response")
276                                         framework.Logf("egressIPSet is: %v", egressIPSet)
277                                         framework.Logf("clientIPSet is: %v", clientIPSet)
278                                         o.Expect(len(clientIPSet)).Should(o.BeNumerically(">", 0))
279                                         o.Expect(AntiDeepEqual(egressIPSet, clientIPSet)).To(o.BeTrue())
~~~

I believe this is a bug, but when we talked about this, the others seemed to disagree. So there's a chance that I'll remove the above part, but for as long as the internal tests are disabled, I prefer to keep that part of the code and defer the final decision to a later point in time.